src/subscription/subscribe.js

```diff
@@ -284,6 +298,8 @@ export function createSourceEventStream(
       );
     });
   } catch (error) {
-    return Promise.reject(error);
+    return error instanceof GraphQLError
+      ? Promise.resolve({ errors: [error] })
+      : Promise.reject(error);
```

Comment:
@taion Not sure that I fully understand this code, but why need different behavior for `Error` and `GraphQLError`?

Reply:
This matches the logic for: https://github.com/graphql/graphql-js/blob/b24cbcde325d6e53788debae9818dd8cddebf9e1/src/subscription/subscribe.js#L114-L124

GraphQL errors resolve the promise as an `ExecutionResult` with just `errors`, while other errors reject the promise.

Reply:
It would be, but is it worth the complexity? What do you think of just adding a comment?